spec/unit/type/file/content_uri_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'spec_helper'
+require 'uri'
+
+describe Puppet::Type.type(:file).attrclass(:content_uri) do
+  include PuppetSpec::Files
+
+  describe "#validate" do
+
+    let(:path) { tmpfile('file_content_uri_validate') }
+    let(:resource) { Puppet::Type.type(:file).new(:path => path) }
+
+    it "should fail if the URI is not a valid URI" do
+      URI.expects(:parse).with('not_a_uri').raises RuntimeError
+      expect(lambda { resource[:content_uri] = 'not_a_uri' }).to raise_error(Puppet::Error)
```

Comment:
The ruby segfault backtrace includes line 14:

```
c:0035 p:0021 s:0142 e:000140 LAMBDA /home/travis/build/puppetlabs/puppet/spec/unit/type/file/content_uri_spec.rb:14 [FINISH]
```
I'm thinking we should remove the `lambda` and just `expect(resource[:content_uri] = 'not_uri')` directly?